lib/core/src/Cardano/Wallet/Primitive/AddressDerivation.hs

```diff
@@ -7,6 +7,7 @@
 {-# LANGUAGE InstanceSigs #-}
 {-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE RoleAnnotations #-}
```

Comment:
hm - never used this before

Reply:
Maybe I'll bring the topic on next week but, we could make more use of `coerce` and `Coercible` from the base package. This allows for coercing between newtypes that have a same (or compatible) underlying representation. So instead of paying the cost of unwrapping and then re-wrapping a newtype, we can just call `coerce` which has no runtime cost as everything is done at compile-time; it's like a special primitive from GHC / base. 

In order to do that, GHC needs some extra `role` annotation to know whether things can be coerced safely or not.